src/charts/line/line-chart.directive.js

```diff
@@ -0,0 +1,287 @@
+        $scope.lineChartId = 'lineChart';
+        if ($scope.config.chartId) {
+          $scope.lineChartId = $scope.config.chartId + $scope.lineChartId;
+        }
+
+        /*
+         * Convert the config data to C3 Data
+         */
+        $scope.getLineData = function (chartData) {
+          var lineData  = {
+            type: 'area'
+          };
+
+          if (chartData && chartData.dataAvailable !== false && chartData.xData) {
+            lineData.x = chartData.xData[0];
+            // Convert the chartData dictionary into a C3 columns data arrays
+            lineData.columns = Object.keys (chartData).map (function (key) {
+              return chartData[key];
+            });
+          }
+
+          return lineData;
+        };
+
+        $scope.getTooltipTableHTML = function (tipRows) {
+          return '<div class="module-triangle-bottom">' +
+            '  <table class="c3-tooltip">' +
+            '    <tbody>' +
+            tipRows +
+            '    </tbody>' +
+            '  </table>' +
+            '</div>';
+        };
+
+        $scope.lineTooltip = function () {
+          return {
+            contents: function (d) {
+              var tipRows;
+              var percentUsed = 0;
+
+              if ($scope.config.tooltipFn) {
+                tipRows = $scope.config.tooltipFn(d);
+              } else {
+                switch ($scope.config.tooltipType) {
+                case 'valuePerDay':
+                  tipRows =
+                    '<tr>' +
+                    '  <td class="value">' +  d[0].x.toLocaleDateString() + '</td>' +
+                    '  <td class="value text-nowrap">' +  d[0].value + ' ' + d[0].name + '</td>' +
+                    '</tr>';
+                  break;
+                default:
+                  tipRows =
+                    '<tr>' +
+                    '  <td class="value text-nowrap">' +  d[0].value + ' ' + d[0].name + '</td>' +
+                    '</tr>';
+                }
+              }
+              return $scope.getTooltipTableHTML(tipRows);
+            },
+            position: function (data, width, height, element) {
+              var center;
+              var top;
+              var chartBox;
+              var graphOffsetX;
+              var x;
+
+              try {
+                center = parseInt(element.getAttribute('x'));
+                top = parseInt(element.getAttribute('y'));
+                chartBox = document.querySelector('#' + $scope.lineChartId).getBoundingClientRect();
+                graphOffsetX = document.querySelector('#' + $scope.lineChartId + ' g.c3-axis-y').getBoundingClientRect().right;
+                x = Math.max(0, center + graphOffsetX - chartBox.left - Math.floor(width / 2));
+
+                return {
+                  top: top - height,
+                  left: Math.min(x, chartBox.width - width)
+                };
+              } catch (e) {
+              }
+            }
+          };
+        };
+
```

Comment:
I believe we can drop these tooltip functions and just rely on the Patternfly default tooltip functionality.